lib/satellite_compensation/middleware.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# Common space for Insights API stuff
+module Insights
+  module API
+    module Common
+      module SatelliteCompensation
+        # Rack middleware to set content-type for Satellite-forwarded requests
+        class Middleware
+          def initialize(app)
+            @app = app
+          end
+
+          def call(env)
+            # Sometimes Satellite forwards the client requests with an empty string
+            # as content-type and Rails does not like it.
+            if env['HTTP_USER_AGENT'] =~ /satellite/i && env['CONTENT_TYPE'] == ''
```

Comment:
I would prefer looking for `foreman_rh_cloud` in some old versions the user-agent was reporting as `Foreman` instead of `Satellite`. The `foreman_rh_cloud ` was always reported as such though.

```suggestion
            if env['HTTP_USER_AGENT'] =~ /foreman/i && env['CONTENT_TYPE'] == ''
```